Add analytics-ui JS package

kangzj opened the pull request:
Discussion: pgWg9p-i-p2

## Proposed changes:

- [x] Created [Automattic/analytics-ui](https://github.com/Automattic/analytics-ui) repo manually as mirror repository


* Add new `@automattic/analytics-ui` JS package for analytics-related UX components to be shared across Automattic products
* Set up package structure with proper configuration files (package.json, composer.json, jsconfig.json)
* Add TypeScript support with tsconfig.json configuration
* Add initial test setup with Jest and placeholder test
* Simplify package exports to only include main entry point
* Fix changelog link template to use correct repository name
* Update pnpm-lock.yaml to include the new package

## Does this pull request change what data or activity we track or use?

No

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Verify that the new package structure is properly set up in `projects/js-packages/analytics-ui/`
* Run `pnpm install` to ensure dependencies are properly installed
* Run `pnpm --filter @automattic/analytics-ui test` to verify tests pass
* Check that TypeScript configuration is correct by reviewing `tsconfig.json`

kangzj committed:
Add analytics-ui JS package

Introduce a new JS package for analytics-related UX components to be shared across Automattic products. This package provides reusable analytics UI components and state management utilities.

kangzj committed:
Add TypeScript configuration to analytics-ui package

Added TypeScript as a dev dependency and configured tsconfig.json to support TypeScript development in the analytics-ui package.

kangzj committed:
Simplify package exports and add placeholder test

Removed unused exports for action-types and state, keeping only the main entry point. Added a placeholder test to satisfy Jest requirements until proper tests are implemented.

kangzj committed:
Fix changelog link template in composer.json

Updated the changelogger link-template to use the correct repository name (analytics-ui instead of jetpack-analytics-ui) to match the mirror-repo configuration.

kangzj committed:
Remove jsconfig.json in favor of tsconfig.json

Since TypeScript configuration has been added, the jsconfig.json is no longer needed. The tsconfig.json now handles the configuration for the package.

adamwoodnz commented on projects/js-packages/analytics-ui/tests/jest.config.cjs:
```diff
@@ -0,0 +1,7 @@
+const path = require( 'path' );
+const baseConfig = require( 'jetpack-js-tools/jest/config.base.js' );
+
+module.exports = {
```
Do we need any of the rest of the config from https://github.com/Automattic/jetpack/blob/a7cf326619cf7280f25906a7bc8fc6c1b9347920/projects/js-packages/charts/tests/jest.config.cjs#L8?

kangzj replied:
I'll leave this one for later as I'm not certain whether we needed those in charts.

anomiex replied:
I suspect you don't need them in charts. Looks like both `@testing-library/jest-dom` and a `ResizeObserver` stub are already handled by stuff in `baseConfig`.

adamwoodnz commented on projects/js-packages/analytics-ui/composer.json:
```diff
@@ -0,0 +1,42 @@
+{
+	"name": "automattic/analytics-ui",
+	"description": "Analytics related UX to be shared across Automattic products.",
```
```suggestion
	"description": "Analytics related UI to be shared across Automattic products.",
```

adamwoodnz commented on projects/js-packages/analytics-ui/package.json:
```diff
@@ -0,0 +1,28 @@
+	"name": "@automattic/analytics-ui",
+	"version": "0.1.0-alpha",
+	"private": true,
+	"description": "Analytics related UX to be shared across Automattic products.",
```
```suggestion
	"description": "Analytics related UI to be shared across Automattic products.",
```

adamwoodnz commented on projects/js-packages/analytics-ui/README.md:
```diff
@@ -0,0 +1,24 @@
+# analytics-ui
+
+Analytics related UX to be shared across Automattic products.
```
```suggestion
Analytics related UI to be shared across Automattic products.
```

adamwoodnz commented on projects/js-packages/analytics-ui/package.json:
```diff
@@ -0,0 +1,28 @@
+	"exports": {
+		".": "./src/index.jsx"
+	},
+	"scripts": {
```
Feels like all that's left is storybook config and build scripts, before there is a full component development env. Worth adding them?

kangzj replied:
Added

adamwoodnz reviewed (commented):
Nice work, I like the direction. A few comments inline.

kangzj committed:
Fix Jest configuration for analytics-ui package

Simplified the Jest configuration to use the base config directly, matching the pattern used in other JS packages like analytics. Added "type": "module" to package.json and updated the test script to use NODE_OPTIONS=--experimental-vm-modules for ES module support.

kangzj committed:
Apply suggestions from code review

Co-authored-by: Adam Wood <1017872+adamwoodnz@users.noreply.github.com>

kangzj committed:
Revert "Fix Jest configuration for analytics-ui package"

This reverts commit afbda2f6c53f5a03f84b67517c3da14a343b674a.

kangzj committed:
Merge remote-tracking branch 'refs/remotes/origin/add/analytics-ui-js-package' into add/analytics-ui-js-package

kangzj committed:
Add Storybook support to analytics-ui package

- Added analytics-ui to the storybook projects list
- Added storybook and typecheck scripts to package.json

kangzj committed:
Apply suggestions from code review

Co-authored-by: Adam Wood <1017872+adamwoodnz@users.noreply.github.com>

kangzj committed:
Merge remote-tracking branch 'refs/remotes/origin/add/analytics-ui-js-package' into add/analytics-ui-js-package

kangzj committed:
Add tsup build configuration to analytics-ui package

- Added tsup.config.ts with build settings matching the charts package
- Updated package.json with build scripts, exports, and required dependencies
- Converted src/index.jsx to src/index.tsx for TypeScript support
- Updated .gitignore to exclude dist/ build output
- Configured for both ESM and CJS output formats with TypeScript declarations

kangzj committed:
Add @automattic/charts as dependency to analytics-ui

anomiex commented on projects/js-packages/analytics-ui/composer.json:
```diff
@@ -0,0 +1,42 @@
+	"require": {},
+	"require-dev": {
+		"automattic/jetpack-changelogger": "@dev",
+		"automattic/phpunit-select-config": "@dev"
```
You could drop the dep on `automattic/phpunit-select-config`, unless the package is going to include phpunit tests somehow.

anomiex commented on projects/js-packages/analytics-ui/composer.json:
```diff
@@ -0,0 +1,42 @@
+		"automattic/jetpack-changelogger": "@dev",
+		"automattic/phpunit-select-config": "@dev"
+	},
+	"autoload": {
+		"classmap": [
+			"src/"
+		]
+	},
```
Again, if there's never going to be any PHP in the package (which seems like a good assumption for a js-package), you could drop this bit.

anomiex commented on projects/js-packages/analytics-ui/package.json:
```diff
@@ -0,0 +1,59 @@
+	},
+	"scripts": {
+		"build": "pnpm run build:prod",
+		"build:prod": "tsup",
```
I note www.npmjs.com/package/tsup says the package is deprecated, and recommends www.npmjs.com/package/tsdown instead.

Probably not in scope for this PR, but it'd be nice if someone would look into whether we should migrate this and charts.

anomiex commented on projects/js-packages/analytics-ui/tsconfig.json:
You might want to look at how `js-packages/components` uses two files, `tsconfig.build.json` and `tsconfig.json`, so only sources get built while everything gets linted and typechecked.

anomiex commented on projects/js-packages/analytics-ui/tsconfig.json:
```diff
@@ -0,0 +1,8 @@
+		"outDir": "dist",
+		"rootDir": "src"
+	},
+	"include": [ "src" ]
```
Note that, if you have tests or stories under `src/`, this will build them into `dist/` too. Which would probably not be what you want.

anomiex reviewed (commented):
Gave it a look from a monorepo perspective, left some comments.

I note this is currently set up to be used only within the monorepo (or directly from the GitHub mirror repo, I guess), it's not being published to npmjs.com. That's fine if that's your intended use case. If it's not what you intend, let me know and I'll point out the changes needed to have it be published. 🙂
